src/main/java/Duke.java

```diff
@@ -1,10 +1,117 @@
-public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+import java.util.Optional;
+import java.io.FileNotFoundException;
+
+import javafx.application.Application;
+import javafx.stage.Stage;
+
+/**
+ * Main class of this application.
+ */
+public class Duke extends Application {
+    private TaskList tasks;
+    private Gui ui;
+    private Storage storage;
+    private boolean isRunning = true;
+
+    /**
+     * Constructs a new `Duke` instance.
+     * This sets up the UI, and loads task data from file.
+     * For the command-line frontend, this attaches a `Scanner` to standard input.
+     */
+    public Duke() {
+        //ui = new Cli();
+        ui = new Gui(this);
+        storage = new Storage();
+        
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (FileNotFoundException e) {
+            ui.startMessage();
+            ui.showSaveNotFoundMessage(storage.STORAGE_SAVE_PATH);
+            ui.endMessage();
+            tasks = new TaskList();
+        } catch (DukeException e) {
+            ui.startMessage();
+            ui.showError(e);
+            ui.endMessage();
+            tasks = new TaskList();
+        }
+    }
+    
+    protected void processCommand(String command) {
+        ui.startMessage();
+        try {
+            Optional<Command> c = new Parser(command).parse();
+            if (c.isPresent()) {
+                Command cmd = c.get();
+                cmd.execute(tasks, ui, storage);
```

Comment:
Excellent OOP implementation on the Command classes to handle different types of commands.